src/main/java/io/odpf/depot/http/request/Request.java

```diff
@@ -1,11 +1,22 @@
 package io.odpf.depot.http.request;
 
+import io.odpf.depot.error.ErrorInfo;
 import io.odpf.depot.http.record.HttpRequestRecord;
 import io.odpf.depot.message.OdpfMessage;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
 
 import java.util.List;
 
 public interface Request {
 
     List<HttpRequestRecord> createRecords(List<OdpfMessage> messages);
+
+    default StringEntity buildEntity(String stringBody) {
+        return new StringEntity(stringBody, ContentType.APPLICATION_JSON);
+    }
+
+    default HttpRequestRecord createAndLogErrorRecord(ErrorInfo errorInfo, int index) {
+        return new HttpRequestRecord((long) index, errorInfo, false, null);
```

Comment:
this should only be createErrorRecord(). it's not doing logging.

Reply:
ack